src/lavinmq/amqp/client.cr

```diff
@@ -194,6 +198,23 @@ module LavinMQ
         end
       end
 
+      private def handle_update_secret(frame : AMQP::Frame::Connection::UpdateSecret)
+        current_user = @user
+        if !current_user.is_a?(Auth::OAuthUser)
+          close_connection(frame, ConnectionReplyCode::NOT_ALLOWED, "update-secret only supported for OAuth authentication")
```

Comment:
A more generic text won't require future changes if we add other authentication mechanisms that supports updating secret.
```suggestion
          close_connection(frame, ConnectionReplyCode::NOT_ALLOWED, "update-secret not supported for current authentication mechanism")
```